spec/filewatch/sincedb_record_serializer_spec.rb

```diff
@@ -8,31 +8,46 @@ module FileWatch
     let(:opts) { Hash.new }
     let(:io) { StringIO.new }
     let(:db) { Hash.new }
-
-    subject { described_class.new(Settings.days_to_seconds(14)) }
+    subject { SincedbRecordSerializer.new(Settings.days_to_seconds(14)) }
 
     context "deserialize from IO" do
       it 'reads V1 records' do
-        io.write("5391299 1 4 12\n")
-        subject.deserialize(io) do |inode_struct, sincedb_value|
-          expect(inode_struct.inode).to eq("5391299")
-          expect(inode_struct.maj).to eq(1)
-          expect(inode_struct.min).to eq(4)
-          expect(sincedb_value.position).to eq(12)
-        end
+        io.write("5391297 1 4 12\n")
+        io.rewind
+        inode_struct, sincedb_value = nil
+        subject.deserialize(io) { |res| inode_struct, sincedb_value = res }
```

Comment:
not sure why we need to move away from the previous construct? it seems the only meaningful change here is the `io.rewind`?

Reply:
@colinsurprenant I suspect that motivation was due to the missed `rewind`. Without `rewind` the test written in old mode success, because no row was loaded and no check done => green.
With the new shape if we don't `rewind` the test fails

Reply:
probably we could use the old format plus adding a record counter and check the `row counter > 0`

Reply:
I understand for the rewind, what I do not understand is why the expectations were moved out of the `subject.deserialize(io) { ... }` block. Using the block syntax is cleaner and avoids having to declare `inode_struct, sincedb_value = nil`.

Reply:
Done, reverted but added a check that the block is executed (what `io.rewind` resolves)

Reply:
ahhh now I get it why the expectations were moved out of the block 🤪  without the rewind the block was never executed and the expectations never actually run.  In any case, the modification you pushed works too! Thanks.